proarc-common/src/main/java/cz/cas/lib/proarc/common/catalog/AlephXServer.java

```diff
@@ -236,10 +234,10 @@ static URI setQuery(URI u, String newQuery, boolean add) throws MalformedURLExce
         }
     }
 
-    static class Criteria {
+    static final class Criteria {
 
         private static class Field {
-            private static Map<String, String> values = new HashMap<String,String>() {
+            private static Map<String, String> values = new ConcurrentHashMap<String,String>() {
```

Comment:
Dík za ty ostatní úpravy. Tohle je poslední věc. ConcurrentHashMap tady určitě není potřeba. Ta stále neřeší problém, že si vlákna budou přepisovat stejnou mapu novými hodnotami a můžou vznikat nekonzistence i když už synchronně. Synchronizace je řešena přímo v commons.configuration. Tady stačí vytvořit snapshot uložením do nestatické mapy a je to vyřešeno. Každé vlákno má pak čerstvý snapshot a může si sním dělat co chce. Viz. třeba Z3950Catalog, 

Další potíž je nyní zrušení nějaké property v konfiguraci za běhu. Současné řešení, dělá jen add, což znamená, že přidá nebo přepíše hodnotu, ale nezruší klíč.

Takže co potřebujeme je `private final Map<String, String> values = new HashMap<>();` a zrušit static u metod.